crates/provider/src/fillers/gas.rs

```diff
@@ -210,8 +157,6 @@ impl<N: Network> TxFiller<N> for GasFiller {
     {
         if tx.gas_price().is_some() || tx.access_list().is_some() {
```

Comment:
Our `GasFiller` automatically treats transaction as legacy if it has an `accessList` set which is not really correct. We also have a test enforcing this, wondering if this should be changed?

Reply:
I believe this is a bug

Reply:
updated in 5697944